### ISSUE TITLE
Кнопка копировать параграф для цитирования

### ISSUE THREAD

**Issue by audetv** [terratensor/kob-library-app#38]
Надо написать, пока обратил внимание, видел были цитирования текста из выдачи

> «Иными словами поросль “экстрасенсов” ныне взращивается для осуществления миссии пастушеских овчарок при разбредающемся «стаде этих скотов». Такого рода миссия иногда обозначается в разного рода “парапсихологических” изданиях, когда в них публикуются призывы типа «все экстрасенсы в такое-то время должны медитировать на избрание Клинтона президентом США»; в России в 1991 г. аналогичное разпоряжение в отношении «медитации на избрание Ельцина» прошло в определённых кругах в устной форме. Причём “культуризм” в области парапсихологии — наиболее эффектная и потому бросающаяся в глаза сторона жизни современного общества индивидуалистов, но это только частный случай более широкого явления»
> ________________________________________________________
> ВП СССР — Мертвая Вода (2015 цензурированная)
> 
> Экстрасенс ★ https://svodd.ru/3sippr1U

Подумал, что можно было бы сделать иконку копировать цитату в буфер (например, рядом с номером параграфа), чтобы проще было копировать и потом вставлять и распространять. Это, что-то простое и возможно не нужное. Оставляю тут, подумать.

**Comment by iprst** [comment]
Отличная идея. 

Только видимо нужно строчку с короткой ссылкой, ибо её надо как-то генерировать, плюс нам неизвестно, есть ли у пользователя желание показывать свой запрос. 

Цитата в кавычках-ёлочках, прочерк, название источника. 

Иконка кнопки должна быть либо стандартная для копирования — «двойная страница», либо стандартная для цитаты «кавычка».

**Comment by iprst** [comment]
Иконка? 
![copy_icon](https://github.com/terratensor/kob-library-app/assets/129882753/dc301357-9e05-49a5-813c-78408169f7c4)

**Comment by audetv** [comment]
Да, я сделал issue и после копирования посмотрел на эту ссылку и подумал, нужна она там?  Если ссылка нужна, надо подумать как это сделать, самое простое это какой-то свой режим. Копирование цитаты, копирование цитаты со ссылкой.

Можно начать с : Цитата в кавычках-ёлочках, прочерк, название источника.

https://fontawesome.com/search?q=Copy&o=r

https://fontawesome.com/search?q=Share&p=2&o=r

Иконку можно выбрать из free иконсетов, самый распространённый fontawesome, да много их. Или взять ту, что вы прикрепили.

**Comment by iprst** [comment]
> Да, я сделал issue и после копирования посмотрел на эту ссылку и подумал, нужна она там? Если ссылка нужна, надо подумать как это сделать, самое простое это какой-то свой режим. Копирование цитаты, копирование цитаты со ссылкой.

Пока сделаем просто цитирование без ссылки. Посмотрим, как пойдёт. Возможно кто-то даст ОС. Мне пока не представляется что и как. 

> Можно начать с : Цитата в кавычках-ёлочках, прочерк, название источника.

Так точно, простота залог здоровья. 

> Иконку можно выбрать из free иконсетов

Да, можно прямо первую, если мы берём «копировать». Вроде это более общепринятый вариант, нежели «цитировать»? Хотя иконки с кавычками мне встречались, эту нарисовал просто чтобы не задумываться об авторских правах.

**Comment by iprst** [comment]
Думаю отдельного issue не заводя — шрифт с номером параграфа нужно бы уменьшить, смело на два или три пункта, и сделать жирным. Так будет более связно с общим дизайном текста. 

При этом отступ после одного экземпляра выдачи можно слегка уменьшить. 

![image](https://github.com/terratensor/kob-library-app/assets/129882753/b1595d71-9c49-4474-9075-8da18424e0f0)

**Comment by audetv** [comment]
> Думаю отдельного issue не заводя — шрифт с номером параграфа нужно бы уменьшить, смело на два или три пункта, и сделать жирным. Так будет более связно с общим дизайном текста.
> 
> При этом отступ после одного экземпляра выдачи можно слегка уменьшить.

изменил оформление 

Начал делать, копирование и пока остановился, застрял. 
Надо как-то обрабатывать теги внутри цитаты - сделать перенос \r\n\ . 
Надо как-то обрабатывать сноски, которые сейчас выглядят как цитата. (ставить круглые скобки?)
Надо как то ставить прочерк - символами подчеркивания. Но надо как сделать, чтобы прочерк на мобильной версии не складывался в 2 строки, пока не знаю как, можно пока никак не делать.

В общем как всегда, простая задача: нажать кнопку и скопировать параграф, превращается в сложный алгоритм.
В работе, пока не тороплюсь )

**Comment by iprst** [comment]
> изменил оформление

Да, теперь всё отлично, нмв

> Надо как-то обрабатывать теги внутри цитаты - сделать перенос \r\n\ 

Теги переноса имеете в виду? 

> Надо как-то обрабатывать сноски, которые сейчас выглядят как цитата. (ставить круглые скобки?)

Поскольку сейчас решение в том, чтобы оставлять в сносках переграфы, они оформлены цитатами. В реальном тексте, если бы он был без сносок, это были бы просто предложения в скобках, без переноса строки, даже если там идут списки или диалоги — существуют нормы оформления всех этих элементов в строку, соответственно в скобках. В идеале, конечно, сделать так. Но поскольку сноски идут как параграфы, и оформлены как цитаты, нужно что-то изобретать. Например фиксированную длину строки с мягкими переносами и табуляцией в начале, типа такого* 

		* Вот пример сноски, как
		если бы мы делали её
		параграфами с оформлением
		а-ля цитаты, при этом
		используя фиксированную 
		длину строки и табуляцию.


Обратите внимание на звёздочку, тоже стандартный приём. Получается не сноска, а врезка, но с тем же логическим функционалом. Правда не помню, кажется парсер ввода на ФКТ режет табуляцию, но можно использовать непереносимый пробел, вот пять штук — `     ` или вот так одна штука `<0xa0>`

Это так, подумать. Я бы в сносках просто убил переносы строки и привёл бы к одной фразе в скобках. 

> Надо как то ставить прочерк - символами подчеркивания. Но надо как сделать, чтобы прочерк на мобильной версии не складывался в 2 строки, пока не знаю как, можно пока никак не делать.

Отбивку можно сделать любым фиксированным количеством, например длиной с бренд «ВП СССР» в начале каждого названия книги. Также можно использовать не подчёркивание, а обычные три звёздочки. Либо вообще ничего не ставить, просто перенос строки и название книги.

**Comment by audetv** [comment]
> Теги переноса имеете в виду?

да, это так обозначается перенос и возврат строки (windows style), типа мягкий перенос.

>  В реальном тексте, если бы он был без сносок, это были бы просто предложения в скобках, без переноса строки, даже если там идут списки или диалоги — существуют нормы оформления всех этих элементов в строку, соответственно в скобках. В идеале, конечно, сделать так. 

Хорошо, попробую так сделать, так возможно будет даже проще, просто вырезать теги из строки сноски и все.
Сделаю текст сноски в скобках. 

> Отбивку можно сделать любым фиксированным количеством, например длиной с бренд «ВП СССР» в начале каждого названия книги. Также можно использовать не подчёркивание, а обычные три звёздочки. Либо вообще ничего не ставить, просто перенос строки и название книги.

Отбивку сделаю, несколько символов подчеркивания, хорошо. Посмотрим, если будет плохо заменим на звездочки, ну, а вообще ничего не ставить проще всего)

**Comment by iprst** [comment]
> да, это так обозначается перенос и возврат строки (windows style), типа мягкий перенос.

Что такое \r\n это я уловил (CLRF в регулярном выражении), я просто не знаю в каком виде это всё хранится внутри записей базы и засомневался что правильно понял слово теги. 

> так возможно будет даже проще, просто вырезать теги из строки сноски и все.

В основном потому и предлагал этот вариант. В обычных книгах это ещё проще, но там нам со сносками заморачиваться не пришлось, поскольку при распознании такие элементы далеко не всегда сохраняют форматирование, и вся болванка текста идёт как получается. А вот в толстых книгах ничего распознавать не нужно, однако со сносками явно намудрили, есть такие портянки на три страницы, что страшно. Зачем? Ибо! 

> Отбивку сделаю, несколько символов подчеркивания, хорошо. Посмотрим, если будет плохо заменим на звездочки, ну, а вообще ничего не ставить проще всего)

Поглядим и сразу понятно будет. Без отбивки тоже нормально.
